backend/src/auth/services/auth_service.py

```diff
@@ -1,82 +1,225 @@
 from typing import Optional
 
+from fastapi import HTTPException
+from fastapi.responses import JSONResponse
 from sqlalchemy.orm import Session
 
-from src.auth.exceptions import (
-    EmailTaken,
-    InvalidPassword,
-    TokenDoesNotExist,
-    UserDoesNotExist,
-    UsernameTaken,
-)
-from src.auth.models import User
-from src.auth.repositories import AuthRepository
 from src.auth.schemas import (
-    GetUser,
-    LoginResponse,
-    RefreshTokenResponse,
-    RegisterResponse,
-    UserCreate,
+    LoginEndpoint,
+    LoginUserModel,
+    RefreshTokenEndpoint,
+    RegisterEndpoint,
+    RegisterUserModel,
 )
 from src.auth.utils import (
-    decode_jwt_token,
     encode_jwt_token,
     hash_password,
     verify_passwords,
 )
-from src.core.exceptions import (
-    DetailedHTTPException,
+from src.common.exceptions import (
     NotAuthenticated,
 )
+from src.common.utils import publish_handler_event
+from src.core.interceptors.auth_interceptor import verify_user_id
+from src.users.exceptions import (
+    EmailTaken,
+    InvalidPassword,
+    InvalidUserData,
+    RegisterFailed,
+    TokenDoesNotExist,
+    UserDoesNotExist,
+    UsernameTaken,
+)
+from src.users.models.user_model import User
+from src.users.repositories import UserRepository
+from src.users.schemas.profile import CreateProfileModel
+from src.users.schemas.user import CreateUserModel
 
 
 class AuthService:
-    def __init__(self, repository: AuthRepository):
+    def __init__(
+        self,
+        repository: UserRepository,
+    ):
         self.repository = repository
 
     def authenticate_user(
         self, username: str, password: str, db: Session
     ) -> Optional[User]:
+        """
+        Authenticates a user by checking the provided username and password against the database.
+
+        Args:
+        - username: The username of the user to authenticate.
+        - password: The password of the user to authenticate.
+        - db: The database session.
+
+        Returns:
+        - The authenticated user.
+        """
+        # Get user from the database by username
```

Comment:
It is not bad approach, but in our app we dont do it so to keep consitstency you should also remove it

Reply:
removed